src/pages/datasets/requests.tsx

```diff
@@ -1,75 +1,45 @@
-      userName: 'Patrick Cason',
-      userId: 19239123,
-      retrieving: '',
-      epsilonCurrent: 30,
-      epsilonAfterChange: 50
-    },
-    {
-      userAvatarURL: 'https://ca.slack-edge.com/T6963A864-UP9PYUFJT-614414739a77-512',
-      userName: 'Hericles Emanuel',
-      userId: 'asdasd19212312392',
-      retrieving: '',
-      epsilonCurrent: 0,
-      epsilonAfterChange: 70
-    }
-  ]
+  if (!datasetsData) return null
 
+  // TODO : Add logic functionality to accept and reject permissions/budget'1s
+  // TODO : Trigger accept and reject modals to onClickAccept and onClickReject
+  // TODO : Hook to PyGrid API
   return (
     <main className="space-y-4">
       <h1 className="text-4xl text-gray-800">Requests</h1>
-      <p className="text-xl font-light text-gray-400">Accept or deny any permissions or privacy budget changes</p>
-      <div className="grid grid-cols-1 gap-6 xl:gap-8 lg:grid-cols-2">
-        <section>
-          <small className="font-semibold tracking-wide text-gray-800 uppercase">Permissions changes</small>
-          <div className="space-y-6 xl:space-y-8">
-            {permissionChanges.map(permission => (
-              <PermissionRequestCard
-                {...permission}
-                key={`permission-card-${permission.dataset}-${permission.tensors}-${permission.userId}`}
-                onClickReason={() => alert('View reason')}
-              />
-            ))}
-          </div>
-        </section>
-        <section>
-          <small className="font-semibold tracking-wide text-gray-800 uppercase">Budget changes</small>
-          <div className="space-y-6 xl:space-y-8">
-            {budgetChanges.map(budget => (
-              <BudgetChangesCard
-                {...budget}
-                key={`budget-card-${budget.userId}-${budget.epsilonAfterChange}`}
-                onClickReason={() => alert('View reason')}
-              />
-            ))}
-          </div>
-        </section>
-      </div>
+      <p className="text-xl font-light text-gray-400">Accept or deny any permissions</p>
+      <section>
+        <small className="font-semibold tracking-wide text-sm text-gray-800 uppercase">Permissions changes</small>
+        <div className="space-y-6 xl:space-y-6 pt-5">
+          {!isLoading &&
+            !error &&
+            requests
+              .filter(x => x.status === 'pending')
+              .sort((a, b) => {
+                return new Date(b.date).getTime() - new Date(a.date).getTime()
+              })
+              .map(permission => (
+                <PermissionRequestCard
+                  {...permission}
+                  tensors={'data.target'}
+                  dataset={datasetsData.find(x => x.id === permission.objectId).name}
+                  retrieving={'requesting tensor'}
+                  key={`permission-card-${permission.objectId}-${permission.userId}`}
+                  onClickAccept={() => acceptRequest(permission.id.toString())}
```

Comment:
Why do `permission.id.toString()`? Seems like `acceptRequest(permission.id)` is enough.